subsystems/TypeHandlerLibrary/build.gradle.kts

```diff
@@ -6,7 +6,10 @@ plugins {
     `java-library`
 }
 
-apply(from = "$rootDir/config/gradle/common.gradle")
+apply(from = "$rootDir/config/gradle/publish.gradle")
```

Comment:
why do we replace this here? 🤔 
I would guess that
a) we used `common.gradle` for something and
b) `publish.gradle` does (should) not duplicate what's in `common.gradle`

Reply:
`common.gradle` provides:
0. repositories for artifact resolving
1. pmd,spotbug,checkstyle etc

`publish.gradle` - provides:
0. `common.gradle` logic 
1. publish to our maven repository logic.

We should publish this subsystem in external repo for exposing this subsystem to single module builds. (aka make it accessable in engineless builds)

Reply:
Also: `publish.gradle` effectively _extends_ `common.gradle` to avoid said overlap. This just means THL goes from being a generic Java project to one that gets published, which sounds about right based on the PR title (we just need to make sure we're on top of whatever added version tag it uses)